### PR TITLE
Tables for alg params - QGIS Database

### PR DESCRIPTION
**Pull request by havatv**
Goal:

Ticket(s): #4429

- [x] Backport to LTR documentation is required

### CODE REVIEW

File: source/docs/user_manual/processing_algs/qgis/database.rst [hunk 1]
```diff
@@ -18,72 +18,112 @@ be created (see eg :ref:`vector_create_stored_connection`).
...
+     - Defines the encoding of the output layer
+   * - **Overwrite**
+     - ``OVERWRITE``
+     - [boolean]
+       
+       Default: True
+     - If the specified database exists, setting this option to
+       ``True`` will make sure that it is deleted and a new
+       database will be created before the layers are added.
+       If this option is ``False``, the layers will be
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/database.rst]:
```suggestion
       If this option is ``False``, the features will be
```

File: source/docs/user_manual/processing_algs/qgis/database.rst [hunk 2]
```diff
@@ -18,72 +18,112 @@ be created (see eg :ref:`vector_create_stored_connection`).
...
+       appended.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/database.rst]:
```suggestion
       appended to the table.
```

File: source/docs/user_manual/processing_algs/qgis/database.rst [hunk 3]
```diff
@@ -18,72 +18,112 @@ be created (see eg :ref:`vector_create_stored_connection`).
...
+     - [boolean]
+       
+       Default: True
+     - Specifies whether to create a spatial index or not
+   * - **Convert field names to lowercase**
+     - ``LOWERCASE_NAMES``
+     - [boolean]
+       
+       Default: True
+     - Convert the field names of the input vector layer
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/database.rst]:
```suggestion
     - Converts the field names of the input vector layer
```

File: source/docs/user_manual/processing_algs/qgis/database.rst [hunk 4]
```diff
@@ -18,72 +18,112 @@ be created (see eg :ref:`vector_create_stored_connection`).
...
+       to lowercase
+   * - **Drop length constraint on character fields**
+     - ``DROP_STRING_LENGTH``
+     - [boolean]
+       
+       Default: False
+     - Should length constraints on character fields be
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/database.rst]: I'm not sure this kind of sentence structure will not mislead our translators. They ar not aware of the context when translating. To me this is a question (without a question mark!), hence I don't really know how I would translate. Let's be simple and coherent with other boolean (or not) description
```suggestion
     - Specifies whether the length constraints on character fields should be
```

File: source/docs/user_manual/processing_algs/qgis/database.rst [hunk 5]
```diff
@@ -18,72 +18,112 @@ be created (see eg :ref:`vector_create_stored_connection`).
...
+       dropped or not
+   * - **Create single-part geometries instead of multi-part**
+     - ``FORCE_SINGLEPART``
+     - [boolean]
+       
+       Default: False
+     - Should the features of the output layer be
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/database.rst]: ditto

File: source/docs/user_manual/processing_algs/qgis/database.rst [hunk 6]
```diff
@@ -18,72 +18,112 @@ be created (see eg :ref:`vector_create_stored_connection`).
...
-``Drop length constraints on character fields`` [boolean]
-  Specifies whether the length constraints on character fields should be dropped
-  or not.
+.. list-table::
+   :header-rows: 1
+   :widths: 20 20 20 40
+   :stub-columns: 0
 
-  Default: *False*
+   * - Name
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/database.rst]:
```suggestion
   * - Label
```

File: source/docs/user_manual/processing_algs/qgis/database.rst [hunk 7]
```diff
@@ -18,72 +18,112 @@ be created (see eg :ref:`vector_create_stored_connection`).
...
+     - Identifier
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/database.rst]:
```suggestion
     - Name
```

File: source/docs/user_manual/processing_algs/qgis/database.rst [hunk 8]
```diff
@@ -98,63 +138,99 @@ be created (see eg :ref:`label_spatialite`).
...
+     - ``TABLENAME``
+     - [string]
+
+       Default: ''
+     - Defines the table name for the imported vector file.
+       If nothing is specified, the layer name will be used.
+   * - **Primary key field**
+
+       Optional
+     - ``tablefield: any``
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/database.rst]:
```suggestion
     - ``PRIMARY_KEY``
```
?

File: source/docs/user_manual/processing_algs/qgis/database.rst [hunk 9]
```diff
@@ -290,20 +395,36 @@ the layer itself.
...
+     - ``SQL``
+     - [string]
+     - Defines the SQL query, for example
+       ``'UPDATE my_table SET field=10'``.
 
 Outputs
 .......
-No new layer is created. The SQL query is executed in place on the layer.
+
+No output is created.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/database.rst]: are you sure that "no new layer" is same as "no output layer"? if you do an update on the layer, you get the same layer with new value. Don't we have case of alg that do no output a new layer? might be worth checking how we manage them and be coherent with them.

File: source/docs/user_manual/processing_algs/qgis/database.rst (hunk 4 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/database.rst]: For boolean variables, "should" is appropriate (perhaps it should be a rule to always start the sentence describing a boolean with "should"?).  The question mark, should be added.

File: source/docs/user_manual/processing_algs/qgis/database.rst (hunk 9 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/database.rst]: "No new layer" is weaker than "No output".  "No new layer" could mean that HMTL is output and can be viewed in the result viewer and that a number of other things are returned from the algorithm, as long as there is no layer returned.
I have checked the algorithm, and the algorithm returns an empty dictionary - so no output is created (that also means that no new layer is created). Coherency is good, and this should be the way we describe this situation (empty dictionary returned).

**Pull request merged**